src/Sign.SignatureProviders.TrustedSigning/TrustedSigningService.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE.txt file in the project root for more information.
+
+using System.Diagnostics;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using Azure;
+using Azure.CodeSigning;
+using Azure.CodeSigning.Models;
+using Azure.Core;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Sign.Core;
+
+namespace Sign.SignatureProviders.TrustedSigning
+{
+    internal sealed class TrustedSigningService : ISignatureAlgorithmProvider, ICertificateProvider, IDisposable
+    {
+        private static readonly SignRequest _emptyRequest = new(SignatureAlgorithm.RS256, new byte[32]);
+
+        private readonly CertificateProfileClient _client;
+        private readonly string _accountName;
+        private readonly string _certificateProfileName;
+        private readonly ILogger<TrustedSigningService> _logger;
+        private readonly SemaphoreSlim _mutex = new(1);
+        private X509Certificate2? _publicKey;
+
+        public TrustedSigningService(
+            IServiceProvider serviceProvider,
+            TokenCredential tokenCredential,
+            Uri endpointUrl,
+            string accountName,
+            string certificateProfileName)
+        {
+            ArgumentNullException.ThrowIfNull(serviceProvider, nameof(serviceProvider));
+            ArgumentNullException.ThrowIfNull(tokenCredential, nameof(tokenCredential));
+            ArgumentNullException.ThrowIfNull(endpointUrl, nameof(endpointUrl));
+            ArgumentException.ThrowIfNullOrEmpty(accountName, nameof(accountName));
+            ArgumentException.ThrowIfNullOrEmpty(certificateProfileName, nameof(certificateProfileName));
+
+            _client = new CertificateProfileClient(tokenCredential, endpointUrl);
+            _accountName = accountName;
+            _certificateProfileName = certificateProfileName;
+
+            _logger = serviceProvider.GetRequiredService<ILogger<TrustedSigningService>>();
```

Comment:
From a DI perspective, this is a bit of an antipattern. Ideally, we would inject the logger and not the service provider itself since it's a god object.

Reply:
I kept this in style with the rest of the project but I do think this is a good question. I think that doing `serviceProvider.GetRequiredService<ILogger<TrustedSigningService>>()` inside the `TrustedSigningService` would make more sense. I can change this and do an update to the other SigningProviders in another PR?

Reply:
That sounds good to me. Ideally `IServiceProvider` is passed to as few places as possible. Any code that receives and `IServiceProvider` can do ANYTHING. So it makes the code harder to understand and predict.

Reply:
I have update the PR to make this change. And I will create another PR to make the changes to the other SigningProviders once this has been merged.